File: DynamicAppConfigIOS/Example/DynamicAppConfig/AppConfig/ExampleAppConfigLogLevel.swift
//
//  ExampleAppConfigLogLevel.swift
//  DynamicAppConfig Example
//
//  App config: application configuration log level
//  An enum used by the application build configuration
//

import UIKit
import DynamicAppConfig

// Enum definition using string raw values for storage
enum ExampleAppConfigLogLevel: String, CaseIterable {
    
    case logDisabled = "logDisabled"
    case logNormal = "logNormal"
    case logVerbose = "logVerbose"

}

File: DynamicAppConfigAndroid/DynamicAppConfigExample/src/main/java/com/crescentflare/dynamicappconfigexample/appconfig/ExampleAppConfigLogLevel.java
package com.crescentflare.dynamicappconfigexample.appconfig;

/**
 * App config: application configuration log setting
 * An enum used by the application build configuration
 */
public enum ExampleAppConfigLogLevel {

    LogDisabled("logDisabled"),
    LogNormal("logNormal"),
    LogVerbose("logVerbose");

    private final String text;

    ExampleAppConfigLogLevel(String text) {
        this.text = text;
    }

    public static ExampleAppConfigLogLevel fromString(String text) {
        if (text != null)
            for (ExampleAppConfigLogLevel e : ExampleAppConfigLogLevel.values())
                if (text.equalsIgnoreCase(e.text))
                    return e;
        return LogDisabled;
    }

    public String toString() {
        return text;
    }
}
